prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -97,7 +98,10 @@ class ReportFunction {
                     report = validatedRequest.report
                     routeReport(context, workflowEngine, validatedRequest, actionHistory)
                     val responseBody = createResponseBody(validatedRequest, actionHistory)
-                    workflowEngine.receiveReport(validatedRequest, actionHistory)
+                    workflowEngine.recordReceivedReport(
+                        report, request.body!!.toByteArray(), validatedRequest.sender!!,
```

Comment:
Rather than just use the not null assert operator here, I'd like you to check for null and then throw a meaningful error. I know that we verify that the body and sender are NOT null in validateRequest, however, using the not null assert just throws a `java.lang.NullPointerException` which is less clear and less readable than something that explains the error. This isn't a breaking review request, just something I feel is important. Any time you find yourself using `!!` in production code it should feel like code smell.

Reply:
Makes sense.  It felt weird to check it again, but I'll add it.